eureka-core/src/main/java/com/netflix/eureka/util/MeasuredRate.java

```diff
@@ -31,30 +31,45 @@
     private static final Logger logger = LoggerFactory.getLogger(MeasuredRate.class);
     private final AtomicLong lastBucket = new AtomicLong(0);
     private final AtomicLong currentBucket = new AtomicLong(0);
+
+    private final long sampleInterval;
     private final Timer timer;
 
-    // private ctor for NO_OP instance
-    private MeasuredRate() {
-        this.timer = null;
-    }
+    private volatile boolean isActive;
```

Comment:
volatile is not necessary here, since both start and stop methods are synchronized and isStarted is used only in these methods.   (synchronized methods force a sync with object's entire state prior to execution)